OSRMTextInstructions/OSRMTextInstructions.swift

```diff
@@ -263,23 +252,23 @@ public class OSRMInstructionFormatter: Formatter {
             }
             
             if scanner.scanString("}", into: nil) {
-                if let tokenType = TokenType(rawValue: token as! String) {
-                    var replacement: String
-                    switch tokenType {
-                    case .wayName: replacement = wayName
-                    case .destination: replacement = destination
-                    case .exit: replacement = exit
-                    case .rotaryName: replacement = rotaryName
-                    case .laneInstruction: replacement = laneInstruction ?? ""
-                    case .modifier: replacement = modifierConstant
-                    case .direction: replacement = directionFromDegree(degree: bearing)
-                    case .wayPoint: replacement = nthWaypoint
-                    }
-                    if tokenType == .wayName {
-                        result += replacement // already modified above
-                    } else {
-                        result += modifyValueByKey?(tokenType, replacement) ?? replacement
-                    }
+                let tokenType = OSRMTokenType(rawValue: token as! String)
+                var replacement: String
+                switch tokenType {
+                case OSRMTokenType.wayName: replacement = wayName
+                case OSRMTokenType.destination: replacement = destination
+                case OSRMTokenType.exit: replacement = exit
+                case OSRMTokenType.rotaryName: replacement = rotaryName
+                case OSRMTokenType.laneInstruction: replacement = laneInstruction ?? ""
+                case OSRMTokenType.modifier: replacement = modifierConstant
```

Comment:
This isn't a good use case for an extensible string enumeration. Please reconsider the suggestions in https://github.com/Project-OSRM/osrm-text-instructions.swift/pull/8#discussion_r102850178.